pyrtools/tools/display.py

```diff
@@ -351,29 +359,40 @@ def check_shape_1d(shapes):
     max_shape = []
     for i, _ in enumerate(images[0].shape):
         max_shape.append(check_shape_1d([img.shape[i] for img in images]))
+    if len(max_shape) > 2 and as_rgb and max_shape[-1] in [3, 4]:
+        # then we have some color images here, so let's not consider the final
+        # dimension (which is RGB(A))
+        max_shape = max_shape[:-1]
     zooms = []
     for img in images:
-        # this checks that there's only one unique value in the list max_shape[i] // img.shape[i],
-        # where i indexes through the dimensions; that is, that we zoom each dimension by the same
-        # amount. this should then work with an arbitrary number of dimensions (in practice, 1 or
-        # 2)
-        if len(set([max_shape[i] // img.shape[i] for i in range(img.ndim)])) > 1:
-            raise Exception("Both height and width must be multiplied by same amount!")
+        # this checks that there's only one unique value in the list
+        # max_shape[i] // img.shape[i], where i indexes through the dimensions;
+        # that is, that we zoom each dimension by the same amount. this should
+        # then work with an arbitrary number of dimensions (in practice, 1 or
+        # 2). by using max_shape instead of img.shape, we will stop when it has
+        # fewer values (as in the case a bit above when `as_rgb` is True)
+        if len(set([s // img.shape[i] for i, s in enumerate(max_shape)])) > 1:
+            raise Exception("Both height and width must be multiplied by same amount but got "
+                            f"image shape {img.shape} and max_shape {max_shape}!")
         zooms.append(max_shape[0] // img.shape[0])
     return zooms, max_shape
 
 
 def imshow(image, vrange='indep1', zoom=1, title='', col_wrap=None, ax=None,
-           cmap=None, plot_complex='rectangular', **kwargs):
-    '''show image(s)
+           cmap=None, plot_complex='rectangular', as_rgb=True, **kwargs):
+    """Show image(s).
 
     Arguments
     ---------
     image : `np.array` or `list`
-        should be a 2d array (one image to display), 3d array (multiple images to display, images
-        are indexed along the first dimension), or list of 2d arrays. the image(s) to be shown.
-        all images will be automatically rescaled so they're displayed at the same size. thus,
-        their sizes must be scalar multiples of each other.
+        should be a 2d array (one image to display), 3d array (multiple
+        grayscale images to display, images are indexed along the first
+        dimension, or a single RGB(A) image, with RGB(A) along the final
+        dimension), list of 2d or 3d arrays (grayscale or RGB(A) images,
+        respectively), or 4d array (multiple RGB(A) images). the image(s) to be
+        shown. all images will be automatically rescaled so they're displayed
+        at the same size. thus, their sizes must be scalar multiples of each
+        other.
```

Comment:
rephrasing: `np.array` or `list` (of np.array). If grayscale, the as_rgb argument (see below) must be set to True else, must be set to False. For grayscale, images are 2d array of shape (h,w), or 3d array of shape (n,h,w) where images are indexed on first dimension. You can also a provide a list of 2d arrays. For RGB(A) images, the image can be a 3d array of shape (h,w,3), 4d array of shape (n, h,w,3) or list of 3d arrays.  --- something like that so that it's emphasized that the as_rgb argument must be set correctly and then clearly delineate sentences for  grayscale vs. rgb sizing.

Reply:
I like that. 

Right now, if `as_rgb=True` and it's a grayscale image, it won't fail, it will properly plot them as grayscale (since that's the default value). Should I make it default to `False` and fail if `as_rgb=True` and it's grayscale? (So, it fails if `as_rgb=False` and it can't figure out how to plot it as grayscale, but not if `as_rgb=True` and it can't figure out how to plot it as RGB(A))

Reply:
I would say it should fail both if `as_rgb=True` and it's a grayscale image, and if `as_rgb=False` and it can't plot it as RGB. Basically the image input and that argument should always match or it should fail.